### PR TITLE
[SQL Server] (No.1) initialize sqlserver libs and define a few interfaces.

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    private String kind;
+    private String administratorLogin;
+    private String administratorLoginPassword;
+    private String version;
+    private String state;
+    private String fullyQualifiedDomainName;
+    private String type;
+
+    private List<Object> sqlFirewallRules;
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java:33]: Shall we create also create a correspond entity for fire wall rule?

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java:33]: typo line.

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public class SqlServerEntity implements IAzureResourceEntity {
+
+    private String name;
+    private String id;
+    private String resourceGroup;
+    private String subscriptionId;
+    private Region region;
+
+    private String kind;
+    private String administratorLogin;
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java:24]: shall we rename it to `administratorLoginName`? as `administratorLogin` is a bit confused

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private String administratorLoginPassword;
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java:25]: Do we need to add admin password in entity?

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    private String resourceGroup;
+    private String subscriptionId;
+    private Region region;
+
+    private String kind;
+    private String administratorLoginName;
+    private String version;
+    private String state;
+    private String fullyQualifiedDomainName;
+    private String type;
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java:28]: for type, state and kind, give the possible values in javadoc

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServerCreator.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    ISqlServerCreator<T> withAdministratorLogin(String administratorLogin);
+
+    ISqlServerCreator<T> withAdministratorLoginPassword(String administratorLoginPassword);
+
+    ISqlServerCreator<T> withEnableAccessFromAzureServices(boolean enableAccessFromAzureServices);
+
+    ISqlServerCreator<T> withEnableAccessFromLocalMachine(boolean enableAccessFromLocalMachine);
+
+    T commit();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServerCreator.java:26]: for commit, this is already an interface in common-lib

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServerCreator.java]: updated.

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java (hunk 4 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/model/SqlServerEntity.java:31]: add TODO.

**Review by andxu — APPROVED** [review]

**Pull request merged**